exercises/src/main/scala/fpinscala/state/State.scala

```diff
@@ -30,34 +30,118 @@ object RNG {
       (f(a), rng2)
     }
 
-  def nonNegativeInt(rng: RNG): (Int, RNG) = ???
+  // 6.1
+  // Use RNG.nextInt to generate a random number between 0 and Int.MaxValue inclusive.  Corner case
+  // when nextInt returns Int.MinValue, which doesn't have a negative counterpart
+  def nonNegativeInt(rng: RNG): (Int, RNG) = {
+    val (i, r) = rng.nextInt
+    if (i < 0) (-i + 1, r) else (i, r)
```

Comment:
I think missed or you misunderstood the corner case above

```
scala> -1 * Int.MinValue
val res3: Int = -2147483648
```

Also, outside of that corner case, you can use `Math.abs(int)`. It might not be best to use here anyway though, since in that corner case:

```
scala> Math.abs(Int.MinValue) == Int.MinValue
val res6: Boolean = true
```